### DISCUSSION TITLE
Module MVC sample

### DISCUSSION THREAD

**Discussion by agriffard** [OrchardCMS/OrchardCore#14809] [Ideas]
Based on this new sample https://github.com/OrchardCMS/OrchardCore.Samples/tree/main/DashboardApplication,
I created https://github.com/agriffard/ModularMvc that is a .NET 8 MVC app with OC 1.8 preview. 

The front urls like `/` or `/Home/Index` are not working.

In program.cs, I commented:
```
app.MapControllerRoute(
    name: "default",
    pattern: "{controller=Home}/{action=Index}/{id?}");
```
in favor of:
`app.UseOrchardCore();`

How can I find the good configuration to make it work?
Do I have to create a specific module just to handle the Home route?

**Comment by MikeAlhayek** [comment]
@agriffard 

You still need to use `app.UseOrchardCore();`. However, you'll need to register the route like this using the OrchardCoreBuilder

```
.Configure((app, routes, services) =>
{
    routes.MapAreaControllerRoute(
        name: "home",
        areaName: "DashboardApplication",
        pattern: "/",
        defaults: new { controller = "Home", action = "Index" });
});
```

Now, you can add a HomeController with Index method in the `DashboardApplication` project and return a view.

**Comment by agriffard** [comment]
Thank you for your answer @MikeAlhayek 

I tried your solution but it does not work. I am stil unable to get something on the `/` url.

**Reply by MikeAlhayek** [reply]
It works for me. I'll submit a PR to show you.

**Reply by MikeAlhayek** [reply]
an update. I did work for me when `DashboardApplication` is not the starting project. It did not work when DashboardApplication is the starting project. We may need to look into OC and see why the web project routes are not discoverable.
